.github/workflows/main.yml

```diff
@@ -112,7 +116,57 @@ jobs:
           # We ignore it when no files could be found because the test command
           # could also be failed because of an unit test. In this case there
           # would be no failed golden test to upload.
-          if-no-files-found: ignore   
+          if-no-files-found: ignore
+  
+  android-integration-test:
+    # Is recommended to run the job with a macOS machine to take advantage of
+    # hardware acceleration support provided by HAXM. See more details in the
+    # README.md of the Android emulator action:
+    # https://github.com/ReactiveCircus/android-emulator-runner#usage
+    runs-on: macos-latest
+    if: ${{ github.event.pull_request.draft == false }}
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+
+      - name: Set Flutter version from FVM config file to environment variables
+        uses: kuhnroyal/flutter-fvm-config-action@v1
+
+      - uses: subosito/flutter-action@main
+        with:
+          flutter-version: ${{ env.FLUTTER_VERSION }}
+          channel: ${{ env.FLUTTER_CHANNEL }}
+      
+      - name: Run integration tests
+        uses: reactivecircus/android-emulator-runner@v2
+        env:
+          USER_1_EMAIL: ${{ secrets.INTEGRATION_TEST_USER_1_EMAIL }}
+          USER_1_PASSWORD: ${{ secrets.INTEGRATION_TEST_USER_1_PASSWORD }}
+        # We are using the "google_atd" target because it's designed for
+        # automated testing.
+        #
+        # See
+        # https://android-developers.googleblog.com/2021/10/whats-new-in-scalable-automated-testing.html.
+        #
+        # Because of the use of "google_atd" we are bit limited in the choice of
+        # API level, arch and channel. See here all requirements:
+        # https://github.com/ReactiveCircus/android-emulator-runner#configurations 
```

Comment:
Interesting question. Can't test this right now, because integration screenshot testing isn't working yet, see https://github.com/flutter/flutter/issues/103222